File: content-solutions/graph-representation/graph-trusted-shop.java
class Solution {
    public int solution(int n, int[][] recs) {
        int[] inn = new int[n + 1];
        int[] out = new int[n + 1];
        for (int[] r : recs) {
            out[r[0]]++;
            inn[r[1]]++;
        }
        for (int x = 1; x <= n; x++) {
            if (inn[x] == n - 1 && out[x] == 0) return x;
        }
        return -1;
    }
}

File: content-solutions/graph-representation/graph-trusted-shop.js
function solution(n, recs) {
  const inn = new Array(n + 1).fill(0);
  const out = new Array(n + 1).fill(0);
  for (const [a, b] of recs) {
    out[a] += 1;
    inn[b] += 1;
  }
  for (let x = 1; x <= n; x += 1) {
    if (inn[x] === n - 1 && out[x] === 0) return x;
  }
  return -1;
}
